File: tests/uasm/sample/boolean_simp/BooleanSimpTest.ts
import { UdonBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonDecorators";
import { UdonSharpBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonSharpBehaviour";
import { Debug } from "@ootr/udon-assembly-ts/stubs/UnityTypes";

@UdonBehaviour()
export class BooleanSimpTest extends UdonSharpBehaviour {
  Start(): void {
    const flag = true as boolean;
    const a: boolean = flag && true;
    const b: boolean = flag || false;
    const c: boolean = !(flag === (false as boolean));
    Debug.Log(a);
    Debug.Log(b);
    Debug.Log(c);
  }
}

File: tests/uasm/sample/boolean_simp/BooleanSimpTest.cs
using UdonSharp;
using UnityEngine;

[UdonBehaviourSyncMode(BehaviourSyncMode.None)]
public class BooleanSimpTest : UdonSharpBehaviour
{
    public void Start()
    {
        bool flag = true;
        bool a = flag && true;
        bool b = flag || false;
        bool c = !(flag == false);
        Debug.Log(a);
        Debug.Log(b);
        Debug.Log(c);
    }
}
